src/include/stir/recon_buildblock/Hessian_row_sum.h

```diff
@@ -0,0 +1,154 @@
+#include "stir/recon_buildblock/PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData.h"
+
+START_NAMESPACE_STIR
+
+class Succeeded;
+
+/*! \ingroup recon_buildblock
+  \brief Implementations of the two spatially variant penalty strength (kappa) computation methods propose by Tsai et.al (2020).
+
+  Two methods of computing the kappa images:
+    - use an approximation of the Hessian of the log-likelihood
+      \code
+        kappa = sqrt[ backproj( forwproj(ones) / y) ]
+      \endcode
+    - use a current image estimate to compute the hessian of the objective function
+    \code
+      kappa = sqrt[ backproj( (y / (forwproj(lambda) + b)^2 ) * forwproj(ones)) +
+                beta * prior.Hessian_times_input(lambda, ones)]
+    \endcode
+  where y is the measured data, lambda the current image estimate, ones an image of ones.
+
+  For more details, see: Tsai, Y.-J., Schramm, G., Ahn, S., Bousse, A., Arridge, S., Nuyts, J., Hutton, B. F.,
+  Stearns, C. W., &  Thielemans, K. (2020). <i>Benefits of Using a Spatially-Variant Penalty Strength With Anatomical
+  Priors in PET Reconstruction</i>. IEEE Transactions on Medical Imaging, 39(1), 11–22.
+  https://doi.org/10.1109/TMI.2019.2913889
+*/
+template <typename TargetT>
+class Hessian_row_sum:
+        public ParsingObject
+{
+public:
+    Hessian_row_sum();
+
+    //! The main function to call to compute spatially variant penalty strength images
+    //! Loads the input image from file, sets up, compute Hessian row sum (with input = ones), sqrts the result,
+    //! and saves it.
+    void process_data();
+
+    //! Saves the output image as output_filename
+    void save_output();
+
+    //! get and set methods for the objective function
+    //@{
+    GeneralisedObjectiveFunction<TargetT > const& get_objective_function();
+    void set_objective_function_sptr(const shared_ptr<GeneralisedObjectiveFunction<TargetT > > &obj_fun);
+    //@}
+
+    //! get and set methods for the input image
+    //@{
+    shared_ptr<TargetT> get_input_image();
+    void set_input_image(shared_ptr <TargetT > const& image);
+    //@}
+
+    //! get method for returning the kappa_image
+    shared_ptr<TargetT> get_output_target_sptr();
+
+    //! This method resets the output target to uniform 0s
+    void reset_output_target_sptr(shared_ptr <TargetT > const& image);
+
+    //! get and set methods for use approximate hessian bool
+    //@{
+    bool get_use_approximate_hessian();
+    void set_use_approximate_hessian(bool use_approximate);
+    //@}
+
+    //! get and set methods for the compute with penalty bool
+    //@{
+    bool get_compute_with_penalty();
+    void set_compute_with_penalty(bool with_penalty);
+    //@}
+
+    //! Computes the objective function Hessian at the current image estimate. Can use the penalty's Hessian
+    //! kappa_image_sptr is the
+    void compute_Hessian_row_sum();
+
+    //! Computes the approximate Hessian of the objective function. Cannot use penalty's Hessian
```

Comment:
> Cannot use penalty's Hessian

Probably best to explain way somewhere. Code should check that the penalty is zero?

Reply:
`accumulate_sub_Hessian_times_input` does the check within it incase penalty is zero.